### ISSUE TITLE
Return json error on non-json body

### ISSUE THREAD

**Issue by pscanf** [staticdeploy/convexpress#13]
[Link to code](https://github.com/staticdeploy/convexpress/blob/2731e16354d8439bb50b482d2528461918402baa/src/parse-body.js#L38)

On non-JSON, `jsonMiddleware` returns an ugly html error with a stack trace, potentially leaking information. Intercept the error and return a `{"message": "Invalid JSON"}` instead.

**Comment by Biuni** [comment]
Can you add more information about this issue? I tried to solve it but I don't display the "ugly html error". Thanks!

**Comment by pscanf** [comment]
Sure! The _ugly html_ is returned by `jsonMiddleware` when for instance you call an endpoint with a malformed json.

### Example

Assuming you have a convexpress server running on `localhost:8080` serving a `/endpoint` route:
```sh
curl -X POST -H "content-type: application/json" -d "{" "localhost:8080/endpoint"
```

You should get:
```html
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Bad Request</pre>
</body>
</html>
```

---

This is because the error is handled by `jsonMiddleware`, which replies with the body above. The idea is to keep using `jsonMiddleware` to detect the error, but to let `convexpress` handle the error, and make it reply with a `{"message": "Invalid JSON"}` (or similar), which is consistent with other error responses.

Thanks for trying to contribute :)

**Comment by pscanf** [comment]
Fixed by #16
